### ISSUE TITLE
Current definition for backbone doesn't allow to set defaults with ES6 classes

### ISSUE THREAD

**Issue by anton-fomin** [flow-typed/flow-typed#2712]
For example the following code 

```
class EntityModel extends Backbone.Model {

  defaults() {
    return {foo: "bar"};
  }

  constructor(attributes?: Backbone$Attrs, options?: ModelOpts): void {
    super(attributes, options);
  }
}
```

Will produce a compilation error
```
Cannot extend property Model [1] with EntityModel because property defaults is read-only in EntityModel [2]
but writable in Model [3].
```

Passing defaults in attributes or options compiles, but doesn't work. 
The same for setting defaults to this in the constructor.
The only way to have default values with ES6 classes is to use defaults() method.

I would suggest to remove this line https://github.com/flow-typed/flow-typed/blob/master/definitions/npm/backbone_v1.x.x/flow_v0.25.x-/backbone_v1.x.x.js#L85
It will leave only defaults as a function in definition. I believe it should not interfere with non ES6 classes syntax as we pass object there.

In other words it will still allow to write for example
```
const EntityModel: Class<IEntityModel> = Backbone.Model.extend({
  defaults: {
  }
})
```

**Comment by AndrewSouthpaw** [comment]
Seems fine to me, make the change and write a few tests (ES6 style and non-ES6 style) to make sure it works. :)
